packages/material-ui/src/LoadingButton/LoadingButton.js

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import withStyles from '../styles/withStyles';
+import capitalize from '../utils/capitalize';
+import Button from '../Button';
+import CircularProgress from '../CircularProgress';
+
+export const styles = () => ({
+  /* Styles applied to the root element. */
+  root: {},
+  /* Styles applied to the root element if `loading={true}`. */
+  loading: {},
+  /* Styles applied to the loadingIndicator element. */
+  loadingIndicator: {
+    position: 'absolute',
+    visibility: 'visible',
+  },
+  /* Styles applied to the loadingIndicator element if `loadingIndicatorPosition="center"`. */
+  loadingIndicatorCenter: {
+    left: '50%',
+    transform: 'translate(-50%)',
+  },
+  /* Styles applied to the loadingIndicator element if `loadingIndicatorPosition="start"`. */
+  loadingIndicatorStart: {
+    left: 10,
+  },
+  /* Styles applied to the loadingIndicator element if `loadingIndicatorPosition="end"`. */
+  loadingIndicatorEnd: {
+    right: 10,
+  },
+  /* Styles applied to the endIcon element if `loading={true}` and `loadingIndicatorPosition="end"`. */
+  endIconLoadingEnd: {
+    visibility: 'hidden'
+  },
+  /* Styles applied to the startIcon element if `loading={true}` and `loadingIndicatorPosition="start"`. */
+  startIconLoadingStart: {
+    visibility: 'hidden'
+  },
+  /* Styles applied to the label element if `loading={true}` and `loadingIndicatorPosition="center"`. */
+  labelLoadingCenter: {
+    visibility: 'hidden',
+  },
+});
+
+const LoadingButton = React.forwardRef(function LoadingButton(props, ref) {
```

Comment:
Some thoughts about naming: As it stands right now React is using "pending" for states where you want to display some sort of spinner. I think this makes more sense here because you don't display a LoadingButton when your content is loading. For these states a skeleton would make more sense. Where you would use this button is when you're transitioning e.g. the screen.

I'm not sure we should necessarily name "PendingButton" ("TransitionButton", "SupsenseButton"?) but at least renaming ´loading` to `pending` should make it more obvious how one should use it. We shouldn't use concurrent mode APIs in our docs yet but later you'd probably use it like

```jsx
function Button() {
  const [startTransition, isPending] = useTransition({timeoutMs: 10000});

  return <LoadingButton onClick={() startTransition(() => {/* do stuff */})} pending={isPending}>Refresh</LoadingButton>
}
```

https://reactjs.org/docs/concurrent-mode-patterns.html#the-three-steps and the concurrent mode docs in general explain this much better.

Reply:
Would something like this then make sense for the API:
```jsx
<BusyButton
  pending
  pendingIndicator="Fetching..."
  pendingIndicatorPosition="center"
>
  Get data
</BusyButton>
```
I was thinking, following `Suspsense` API to add `fallback` prop, but here it is not really a fallback, but more of a spinner, or some kind of pending indication, so I think `pendingIndicator` would make more sense..